### DISCUSSION TITLE
OC Headless + SSG (like Astro)

### DISCUSSION THREAD

**Discussion by agriffard** [OrchardCMS/OrchardCore#12664] [Ideas]
I recently tried [Astro](https://astro.build/) and I really like the provided features, even if I am not a good frontend js developer.

There could be a way to mix OC with this kind of application, calling the GraphQL api with an access token in order to retrieve specific contents (menu, pages, blog posts, custom contents).

Here are a few pointers and Astro examples using another Headless api, Storyblok:
Astro docs: https://docs.astro.build/en/guides/data-fetching/#example-storyblok-api
Demo: https://stackblitz.com/edit/astro-sdk-demo (Add `/test` after the `webcontainer.io` url on the upper right to see an example)

A SDK is available to integrate easily the api.
Storyblok JavaScript SDK: https://github.com/storyblok/storyblok-js
Storyblok package for Astro : https://github.com/storyblok/storyblok-astro

Is anyone interested in Static site generators fetching content from Orchard Core Headless like this?

**Comment by agriffard** [comment]
Let's say that I have a Static Site Generator fetching data from an api exposed a Headless Orchard Core instance and that the site is built by GitHub Actions,

There is currently no WebHook feature in OC that I know of.
If there was, I am right if I imagine that I could declare a webhook that triggers my CI when some content is published?

Does anyone have such an example using an existing other SSG + CMS framework working together with a CI action triggered by a webhook?

Is a Workflow able to do such action?

**Reply by Skrypt** [reply]
If I understand correctly you would generate the website each time a page is added. That could potentially cause issues if 2 pages are added in a really short time. Then, that would trigger the CI twice to do a Site Generation and then subsequent deployments.

I would rather have a button on my website to trigger a Site Generation so that is not controlled by some event that could be triggered quite often.

**Reply by agriffard** [reply]
Ok, thank you for your reply but what kind of url this button would call? 
GitHub would provide one somewhere in the settings?

**Reply by Skrypt** [reply]
I mean from the Orchard Core instance I would generate the website. Then, you can submit the files where needed.
